lib/dynamodb.js

```diff
@@ -1228,176 +1340,158 @@ DynamoDB.prototype.range = function (model, id, start, end, callback) {
+              break;
+          case 'nlike':
+              returnExpression += "(NOT contains(#" + key + "," + getExpressionValueToken(conditionValue.nlike) + "))";
+              break;
+          case 'or':
+          case 'and':
+              throw new Error('and and or conditions are not supported');
+        }
       });
-      return tokens.join(',');
+    } else {
+      //Simple case -- do equivalency in query/scan notation.
+      prepExpressionAttributeName();
+      returnExpression += "(#" + key + " = " + getExpressionValueToken(conditionValue) + ')';
     }
+
+    return returnExpression;
   }
 
-  //The where definition is a complex and not just an equivalancy operation.
-  if (Object.prototype.toString.call(conditionValue) == "[object Object]") {
-    var operators = Object.keys(conditionValue);
-    operators.forEach(function (keyValue) {
-      var useIndex = isIndexed(key, keyValue, conditionValue[keyValue], index);
-      switch (keyValue) {
-      case 'inq':
-        if (useIndex) {
-          addKeyCondition(key, "EQ", conditionValue.inq[0]);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " IN (" + getExpressionValueToken(conditionValue.inq) + ")" + ")";
-        }
-        break;
-      case 'gt':
-        if (useIndex) {
-          addKeyCondition(key, "GT", conditionValue.gt);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " > " + getExpressionValueToken(conditionValue.gt) + ")";
-        }
-        break;
-      case 'gte':
-        if (useIndex) {
-          addKeyCondition(key, "GE", conditionValue.gte);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " >= " + getExpressionValueToken(conditionValue.gte) + ")";
-        }
-        break;
-      case 'lt':
-        if (useIndex) {
-          addKeyCondition(key, "LT", conditionValue.lt);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " < " + getExpressionValueToken(conditionValue.lt) + ")";
-        }
-        break;
-      case 'lte':
-        if (useIndex) {
-          addKeyCondition(key, "LE", conditionValue.lte);
-        } else {
-          prepFilterExpression();
-          params.FilterExpression += "(#" + key + " <= " + getExpressionValueToken(conditionValue.lte) + ")";
-        }
-        break;
-      case 'between':
-        if (useIndex) {
-          addKeyCondition(key, "BETWEEN", conditionValue.between[0], conditionValue.between[1]);
+  let analyzeQueryWhere = (params, queryWhere) => {
+    let returnExpression = '';
+    let properties = Object.keys(queryWhere);
+    properties.forEach((key) => {
+      let conditionValue = queryWhere[key];
+      if (key === 'and') {
+        conditionValue.forEach((whereClause) => {
```

Comment:
For these couple of .forEach() loops, would it be be better to append to an array, and then do array.join(' AND ').